pkg/kubelet/cm/types.go

```diff
@@ -0,0 +1,65 @@
+// +build linux
+
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cm
+
+// QosClass defines the supported QoS classes of Pods/Containers.
+type QosClass string
+
+const (
+	// GuaranteedQoS is the Guaranteed QoS class.
+	GuaranteedQoS = "Guaranteed"
+	// BurstableQoS is the Burstable QoS class.
+	BurstableQoS = "Burstable"
+	// BestEffortQoS is the BestEffort QoS class.
+	BestEffortQoS = "BestEffort"
+)
+
+// ResourceConfig holds information about all the supported cgroup resource parameters.
+type ResourceConfig struct {
+	// Memory limit (in bytes).
+	Memory int64 `json:"memory"`
```

Comment:
Sounds Good. I will have ResourceList internally. I agree that in the future we might want to add more resources, so it makes sense to have ResourceList. Thanks!